functorch/_src/aot_autograd.py

```diff
@@ -409,14 +416,16 @@ def forward(ctx, *deduped_flat_tensor_args):
                 CompiledFunction.compiled_fw, deduped_flat_tensor_args
             )
             num_outs = CompiledFunction.num_outs
-            ctx.save_for_backward(*fw_outs[num_outs:])
+            num_symints = CompiledFunction.num_symints
+            ctx.save_for_backward(*fw_outs[num_outs:-num_symints])
+            ctx.symints = fw_outs[-num_symints:]
             return tuple(fw_outs[0:num_outs])
 
         @staticmethod
         @disable_torchdynamo
         def backward(ctx, *flat_args):
             contiguous_args = [t.contiguous() for t in flat_args]
-            all_args = list(ctx.saved_tensors) + list(contiguous_args)
+            all_args = list(ctx.symints) + list(ctx.saved_tensors) + list(contiguous_args)
```

Comment:
this actually seems to work, since we don't currently optimize to only save a symint once if it's both returned by func and saved for backward.   but i'm still thinking i should rewrite it some better way.